library/std/src/path.rs

```diff
@@ -1074,6 +1074,7 @@ impl FusedIterator for Ancestors<'_> {}
 // When attribute privacy is implemented, `PathBuf` should be annotated as `#[repr(transparent)]`.
 // Anyway, `PathBuf` representation and layout are considered implementation detail, are
 // not documented and must not be relied upon.
+#[cfg_attr(not(any(bootstrap, test)), rustc_diagnostic_item = "PathBuf")]
```

Comment:
The reason these are warning about unused attributes is that the exact same attribute is already defined 10 lines above. I don't remember the implementation details of the `rustc_diagnostic_item` attribute, but I may have been lazy and not added a custom error for multiple attributes, and instead just looked for the first one and marked that as used.

Reply:
Well...that's a little embarrassing. 🤦 But now I know to look for attributes above non-doc comments.

Reply:
@oli-obk Would this be a good use for `delay_span_bug`?

Reply:
iirc there's a `find` somewhere that we should turn into a full for loop instead and report errors after the first one.

Reply:
I mean... technically the current warning is correct. it's just not very helpful

Reply:
Like this? https://github.com/camsteffen/rust/commit/dfe0399c574b976ac87e4dc4e64a1ac92e4fee99